### PR TITLE
add support for docker build context path

### PR DESCRIPTION
**Pull request by adashen**
1. now user could specify the docker build context path at "module.json" at "image/contextPath". The path value could be absolute or relative to the current "module.json"
2. Also refactored the code to define the class "buildsettings"
3. @LazarusX some change in intellisense, please have a check

### CODE REVIEW

File: src/common/utility.ts [hunk 1]
```diff
@@ -267,32 +267,41 @@ export class Utility {
...
         );
     }
 
+    public static getBuildSettings(
+        modulePath: string,
+        dockerFilePath: string,
+        buildOptions?: string[],
+        contextPath?: string): BuildSettings {
+        const optionArray = (buildOptions && buildOptions instanceof Array) ? buildOptions : undefined;
+        const context = contextPath ? path.join(modulePath, contextPath) : path.dirname(dockerFilePath);
```
Comment (LazarusX) [inline src/common/utility.ts]: I think `path.resolve` is better than `path.join` here as `path.resolve` can handle the case where `contextPath` is absolute.

Comment (adashen) [reply src/common/utility.ts]: good catch, will update

**Comment by LazarusX** [comment]
I have verified the IntelliSense features. Looks good to me.

File: src/common/buildSettings.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export class BuildSettings {
+    public readonly options?: string[];
+    public readonly contextPath: string;
+    public readonly dockerFile: string;
+
+    constructor(dockerFile: string, contextPath: string, options?: string[]) {
```
Comment (formulahendry) [inline src/common/buildSettings.ts]: Could simplify with `constructor(public readonly dockerFile: string, public readonly contextPath: string, public readonly options?: string[])`

**Pull request merged**